sdk/core/azure_core/benches/benchmarks.rs

```diff
@@ -20,10 +25,183 @@ fn url_parsing_benchmark(c: &mut Criterion) {
     }
 }
 
+fn http_request_creation_benchmark(c: &mut Criterion) {
```

Comment:
1. I don't understand why we're perf-testing what amounts to allocations.
2. Lots of the code below is not an efficient way of doing this. You should look at generated code (perhaps not always the most efficient, but...) so we get comparative metrics.

For example, you create a string and modify that, but we do this with `Url` which, in many cases, is more efficient (might pre-allocate room if pushing multiple headers).